packages/app/src/components/DBRowTable.tsx

```diff
@@ -761,6 +770,14 @@ export function DBSqlRowTable({
     return { ...config, select, additionalKeysLength };
   }, [primaryKey, partitionKey, config]);
 
+  const {
+    explainResults,
+    isLoading: isExplainLoading,
+    error: explainError,
+  } = useExplainQuery(mergedConfig ?? config, {
```

Comment:
A pattern I've found to make components more reusable is to have the querying hooks be its own component, this way we can always pass in a config object into any component, and have it render the results that we want (similar to `SearchTotalCount` component for example, or for any of our chart components).

The intent here is that if we want to show estimates elsewhere (ex in a chart editor), it'd just be a few lines to slap the component in.

Reply:
This already is it's own component `useExplainQuery.tsx` (https://github.com/hyperdxio/hyperdx/pull/538/files/ec937e1522271aadfa9e77fbb80084ef65942cf4#diff-cb81c6da9e9cb0767375c8b8d50af677205f01406ebb236320d47a37ab10b9a6)

Or am I missing the intent here?

Reply:
Sorry I realize that my comment wasn't super specific, the idea carried from before is that components that render a query from CH should all take in a config object that specifies what kind of query we want from the component, and the component is both in charge of dispatching the query and rendering the results. (Underneath the component can do it in two steps to separate fetching from display, like the DBRowTable sort of does). But the high level API will look like `<MyNewComponent config={myConfig} {...otherDisplayRelatedOptions} />`

This is because most components will either generate or already have a config lying around, and it's very easy to then slot in new UI components like lego blocks into the page by just passing it a config, and not needing to worry about how the component is going to fetch the data.

So as an example SearchTotalCount looks like:
https://github.com/hyperdxio/hyperdx/blob/ec937e1522271aadfa9e77fbb80084ef65942cf4/packages/app/src/DBSearchPage.tsx#L111-L116

which takes in a config and an additional parameter (for caching reasons, might be a bit of a leaky abstraction there!)

This makes it easy if we want to for example slot this into a side panel, dashboard page, or anywhere else, as long as we have a config object around it's a really simple call from the perspective of the feature builder. (Ex. the service dashboard page just has to specify a bunch of config objects and we magically get charts queried and rendered from a few objects): https://github.com/hyperdxio/hyperdx/blob/ec937e1522271aadfa9e77fbb80084ef65942cf4/packages/app/src/ServicesDashboardPage.tsx#L297-L363

Reply:
TL;DR: Right now I have the hook being called  and am passing the results down to render and what you're saying is that the component should both query and render. So instead of the results being passed into the component, pass in a config so it can do the querying as well.

Fixed in this commit.